src/multivariate/mvnormalcanon.jl

```diff
@@ -56,26 +65,17 @@ const ZeroMeanIsoNormalCanon{Axes}  = MvNormalCanon{Float64,ScalMat{Float64},Zer
 
 
 ### Constructors
-function MvNormalCanon(μ::AbstractVector{T}, h::AbstractVector{T}, J::AbstractPDMat{T}) where {T<:Real}
-    length(μ) == length(h) == dim(J) || throw(DimensionMismatch("Inconsistent argument dimensions"))
-    if typeof(μ) === typeof(h)
-        return MvNormalCanon{T,typeof(J),typeof(μ)}(μ, h, J)
-    else
-        return MvNormalCanon{T,typeof(J),Vector{T}}(collect(μ), collect(h), J)
-    end
-end
-
-function MvNormalCanon(μ::AbstractVector{T}, h::AbstractVector{T}, J::AbstractPDMat) where {T<:Real}
+function MvNormalCanon(μ::AbstractVector{T}, h::AbstractVector{T}, J::P) where {T<:Real, P}
     R = promote_type(T, eltype(J))
-    MvNormalCanon(convert(AbstractArray{R}, μ), convert(AbstractArray{R}, h), convert(AbstractArray{R}, J))
+    MvNormalCanon{T,P,typeof(μ)}(convert(AbstractArray{R}, μ), convert(AbstractArray{R}, h), convert(AbstractArray{R}, J))
```

Comment:
For the old function here (relaxed to `AbstractMatrix`, see above):
```suggestion
    MvNormalCanon(convert(AbstractArray{R}, μ), convert(AbstractArray{R}, h), convert(AbstractArray{R}, J))
```